### ISSUE TITLE
Download creates corrupted files on UDF partition, freezes client

### ISSUE THREAD

**Issue by DanMan** [ValveSoftware/steam-for-linux#5767]
#### Your system information

* Steam client version: Sept 8, 2018
* Distribution: Ubuntu 18.04
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes
* udftools: v2.0-2

#### Please describe your issue in as much detail as possible:

Tried to download MGS5 onto a newly created [UDF](https://en.wikipedia.org/wiki/Universal_Disk_Format) 2.01 partition on a new HDD, and whenever it comes to create certain (it seems) files, they end up corrupted. Last time i tried, it was so bad that the whole Steam client froze right after spamming stuff into my kernel ring buffer (see [here](https://github.com/pali/udftools/issues/25)); I couldn't even kill -9 Steam. The files in question are now completely broken. Trying to access them in any way just gives a generic I/O error, preventing me from even deleting them. All I can do is wipe the whole 2TB disk and start over, because it also leaves the partition in an inconsistent, non-fixable state.

#### Steps for reproducing this issue:

1. Setup a Steam library on a UDF partition/drive
2. Try downloading "Metal Gear Solid 5"

No such problems on an ext4 partition on the same drive. It just works. I wonder why it depends on the user software, if the writing will succeed or not. Doesn't the Steam client call the same system routines as any other software?

**Comment by kisak-valve** [comment]
Hello @DanMan, are you still experiencing this issue on an up to date system?

**Comment by DanMan** [comment]
Ok, so I've just tried it again on this [newer Fedora 33 system](https://gist.githubusercontent.com/DanMan/df94fe2dc319aad76fc097c2f90e6329/raw/1ba299c1fbc6bcf3c8cb83f5fa2dc65855cde25b/sysinfo.txt) and I did not run into that error again.

But 2 additional things are different this time: I used an internal HDD instead of a USB attached one, and I used Gnome Disks to create the UDF filesystem instead of [the script that maximizes cross-OS compatibility](https://github.com/JElchison/format-udf). I can test the latter still, if you want but I don't have an empty USB HDD right now.
